src/diagnostics/LogPointsManager.ts

```diff
@@ -0,0 +1,190 @@
+import * as util from 'util';
+import * as vscode from 'vscode';
+import { ISetLogpointResponse } from './logPointsClient';
+import { RemoteScriptSchema } from './remoteScriptDocumentProvider';
+import { Logpoint } from './structs/Logpoint';
+import { LogpointsCollection } from './structs/LogpointsCollection';
+
+class DebugSessionManager {
+    private _logpointsCollectionMapping: { [documentUri: string]: LogpointsCollection };
```

Comment:
we're going to need to account for cases where the service gets restarted & all this state is invalidated.

Reply:
Added a task to track it for now.